ui-speedspacechart/src/components/utils.ts

```diff
@@ -326,26 +326,40 @@ export const interpolate = (x1: number, y1: number, x2: number, y2: number, x: n
 export const clamp = (value: number, min: number, max: number) =>
   Math.min(Math.max(value, min), max);
 
-/** Filter stops to avoid overlapping draw */
+/** Filter stops to avoid overlapping, showing the most important stops first */
 export const filterStops = (
-  stops: LayerData<string>[],
+  stops: LayerData<OperationalPoints>[],
   ratioX: number,
   width: number,
   maxPosition: number,
   minSpace = 8
 ) => {
-  let lastDisplayedPosX: number | null = null;
+  // Sort stops by decreasing weight to prioritize the heaviest ones
+  const sortedStopsByWeight = stops
+    .slice()
+    .sort((a, b) => (b.value.weight ?? 0) - (a.value.weight ?? 0));
```

Comment:
If `stops` is already an array why do we need to use `slice`?

Reply:
The use of .slice() in this context ensures that the original array is not mutated by the .sort() operation. Keeping it ensures the function behaves predictably and doesn't cause unexpected side effects

Reply:
I just discovered this method [toSorted](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted), which is the immutable version of sort. It will help avoid using slice.